modules/network/net_resource.go

```diff
@@ -383,7 +383,7 @@ func configWG(localResource *modules.NetResource, network *modules.Network, wgPe
 // localResource return the net resource of the local node from a list of net resources
 func (n *networker) localResource(resources []*modules.NetResource) *modules.NetResource {
 	for _, resource := range resources {
-		if resource.NodeID.ID == n.nodeID.Identity() {
+		if resource.NodeID.ID == n.identity.NodeID().Identity() {
```

Comment:
Why u have to do this dynamically? the node identity is unlikely to change while the node is running !

Reply:
well I have to get the ID from somewhere

Reply:
Although you are right that the node ID should never change. I think now that we have the identity module we should try to rely on it as much as possible for these kind of things.

Imagine the following scenario:
One day we have to change the format of the node ID or any other case where the node ID change.
If you have in memory copy of the node ID in every module, it is imposible to make sure this new ID will be used properly everywhere. While if you always rely on identityd, then we know this is the only module that needs to reload the ID.